### PR TITLE
js3-w3-niloufar exercises and repoapp

### PR DESCRIPTION
**Pull request by niloufrab**

### CODE REVIEW

File: Week3/homework/js-exercises/ex2.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+//parent class
+class creature {
```
Comment (nabudaldah) [inline Week3/homework/js-exercises/ex2.js:2]: Usually, classes have UpperCase naming ... so "creature" should be "Creature" ... minor adjustment :)

File: Week3/homework/js-exercises/ex2.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  constructor (name, age, location, job  ) {
+    //use super keyword in order to call parent constructor
+    super (name , age);
+    this._location = location;
+    this._job = job;
+
+  }
+
+  possession(){
+    return `${this.name} has a wife , 3 children and a horse `
```
Comment (nabudaldah) [inline Week3/homework/js-exercises/ex2.js:30]: Is "wife and 3 children" etc hard-coded here? Culturally, I would not call a husband/wife or kids "posessions" haha, but that's technically not wrong.

File: Week3/homework/js-exercises/ex2.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+class horse extends creature {
+  constructor (name, age, color){
+    super (name, age);
+    this.color = color;
+  }
+
+
+  activity(){
+    super.activity();
+    console.log('grass and helps transport materials')
```
Comment (nabudaldah) [inline Week3/homework/js-exercises/ex2.js:56]: The console.log() should be removed here...

File: Week3/homework/js-exercises/ex2.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    //use super keyword in order to call parent constructor
+    super.feel();
+  }
+}
+
+//instantiate object
+const abdulkareem = new person('Abdulkareem',35,'Riyadh','constructor');
+const adel = new horse('Adel',15,'brown');
+console.log(adel)
+console.log(abdulkareem)
```
Comment (nabudaldah) [inline Week3/homework/js-exercises/ex2.js:70]: How does this work? The exercise asks for a print-line with all info, here you only print the object. Could you redo?

File: Week3/homework/js-exercises/repo-app3/util/getRepoInfo.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+export function getRepoInfo() {
+  fetch(`https://api.github.com/repos/HackYourFuture/${select.value}`)
+    .then((res) => res.json())
+    .then((data) => {
```
Comment (nabudaldah) [inline Week3/homework/js-exercises/repo-app3/util/getRepoInfo.js:5]: Why not use await here?

File: Week3/homework/js-exercises/repo-app3/util/paginate.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    let contributorElement = document.createElement('div');
+    contributorElement.classList.add(contributor);
+    contributorElement.innerText = contributor;
+
+   
+    wrapper.appendChild(contributorElement);
+  }
+}
+
+export function setupPagination(contributors, wrapper, contributorsPerPage) {
```
Comment (nabudaldah) [inline Week3/homework/js-exercises/repo-app3/util/paginate.js:21]: Why put all functions into one .js script?

**Comment by niloufrab** [comment]
modified repo app

File: Week3/homework/js-exercises/ex2.js (hunk 2 above)
Comment (niloufrab) [reply Week3/homework/js-exercises/ex2.js:30]: right! actually was so complicated how to write this relations.

File: Week3/homework/js-exercises/repo-app3/util/getRepoInfo.js (hunk 5 above)
Comment (niloufrab) [reply Week3/homework/js-exercises/repo-app3/util/getRepoInfo.js:5]: I'm more confident with fetch than async, not specific reason.

File: Week3/homework/js-exercises/repo-app3/util/paginate.js (hunk 6 above)
Comment (niloufrab) [reply Week3/homework/js-exercises/repo-app3/util/paginate.js:21]: thanks, i will fix that.

**Pull request closed without merging**